rust-arroyo/examples/transform_and_produce.rs

```diff
@@ -33,7 +34,7 @@ fn reverse_string(value: KafkaPayload) -> Result<KafkaPayload, InvalidMessage> {
         value.headers().cloned(),
         Some(result_str.to_bytes().to_vec()),
     );
-    Ok(result)
+    Ok(message.replace(result))
```

Comment:
I wonder if we should have the `RunTask` strategy call `message.replace(..)` automatically, and leave the return value here. Wdyt?

Reply:
Let me see what that might look like. @untitaker had some borrow checker concern I think